### DISCUSSION TITLE
How to run something like `pnpm install --recursive <monorepo-package>`

### DISCUSSION THREAD

**Discussion by zachkirsch** [pnpm/pnpm#8682] [Q&A] [answered]
I have a monorepo where one of the packages is `server`. Also relevant, `server` depends on other monorepo packages, e.g. `shared`.

 My current dockerfile is something like:

```dockerfile
COPY . .
RUN pnpm install
```

Is there a way to just install the dependencies for the `server` package (and the other monorepo packages that it depends on)?

I tried `pnpm --filter=server install` and `pnpm --filter=server --recursive install`. That installs dependencies for `server`, but it doesn't install dependencies for the other monorepo packages that `server` depends on (e.g. `shared`).

**Comment by xty** [comment, marked as answer]
> Is there a way to just install the dependencies for the server package (and the other monorepo packages that it depends on)?

`pnpm --filter server... install` might be what you're looking for. Just append an ellipsis (`...`) to the package name.

Also, you can read more about filtering [here in the docs](https://pnpm.io/filtering). The selector syntax is actually quite powerful.

**Reply by zachkirsch** [reply]
Wow TIL, thanks!!
